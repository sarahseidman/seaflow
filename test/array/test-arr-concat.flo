int[] a = [4,5,6];
int[] b = [7,8,9];

int[] c = a + b;

printi(c.length);

printi(c[0]);
printi(c[5]);

char[] d = ['a','b','c'];
char[] e = ['d','e','f'];
char[] f = d + e;
printi(f.length);
printc(f[0]);
printc(f[5]);

float[] fl = [1.3, 2.4, 5.6] + [0.3, 4.5];
printi(fl.length);
printf(fl[4]);

struct Test {
    int x;
    float y;
    char[] name;
};

struct Test one = {1,4.5, "hello"};
struct Test two = {3, 6.5, "world"};
struct Test three = {2,3.5, "good"};
struct Test four = {8, 16.5, "morning"};

struct Test[] t = [ one, two ];
struct Test[] t2 = [ three,four ];
struct Test[] t3 = t + t2;

printi(t3.length);
struct Test res = t3[2];
prints(res.name);