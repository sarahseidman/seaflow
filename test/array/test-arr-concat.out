6
4
9
6
a
f
5
4.500000
4
good